code/modules/arcade/mob_hunt/mob_avatar.dm

```diff
@@ -12,17 +12,22 @@
 	var/list/clients_encountered = list()		//tracks who has already interacted with us, so they can't attempt a second capture
 	var/image/avatar
 
-/obj/effect/nanomob/New(loc, datum/mob_hunt/new_info)
-	..()
+/obj/effect/nanomob/Initialize(mapload, datum/mob_hunt/new_info)
+	. = ..()
 	if(!new_info)
-		qdel(src)
-		return
+		return INITIALIZE_HINT_QDEL
 	mob_info = new_info
 	update_self()
 	forceMove(mob_info.spawn_point)
 	if(!mob_info.is_trap)
 		addtimer(CALLBACK(src, .proc/despawn), mob_info.lifetime)
 
+/obj/effect/nanomob/Destroy()
+	if(SSmob_hunt)
```

Comment:
Quick nitpick that needs doing in another PR.

These `if(SSthingy)` checks are entirely redundant. 
Subsystems are instanced before `world/New()` has even been entered. They exist for the entire game unless an admin manually deletes them, which makes me slap them anyways.

Reply:
Fair enough, I know that SSticker always exists but I wasn't sure about the others. Should I remove this check then or just leave it in?

Reply:
Please remove it so I dont have to at a later date